src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -335,10 +335,8 @@ internal static void PopulatePropertiesFastPath(object obj, JsonTypeInfo jsonTyp
 
                 jsonTypeInfo.OnSerializing?.Invoke(obj);
 
-                List<KeyValuePair<string, JsonPropertyInfo>> properties = jsonTypeInfo.PropertyCache!.List;
-                for (int i = 0; i < properties.Count; i++)
+                foreach (JsonPropertyInfo jsonPropertyInfo in jsonTypeInfo.PropertyCache)
```

Comment:
This set of changes removes dependencies on the bespoke `JsonPropertyDictionary<T>` type in `JsonTypeInfo` caches, using a regular dictionary instead.